### ISSUE TITLE
[typedef] variable-declaration-ignore-function from TSLint not implemented

### ISSUE THREAD

**Issue by niels013** [typescript-eslint/typescript-eslint#1555]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/typedef": [ "error", {
            "arrayDestructuring": true,
            "arrowParameter": true,
            "memberVariableDeclaration": true,
            "objectDestructuring": true,
            "parameter": true,
            "propertyDeclaration": true,
            "variableDeclaration": true
      }]
  }
}
```

```TS
const doSomething = (): void => {
	console.log('Doing something');
};

doSomething();
```

**Expected Result**
No error from ESLint

**Actual Result**
```
...\src\index.ts
  1:7  error  expected doSomething to have a type annotation  @typescript-eslint/typedef
```
**Additional Info**
When I change the code to
```TS
const doSomething: () => void = (): void => {
	console.log('Doing something');
};

doSomething();
```
the code passes linting.
When I check the rule page of TSLint (https://palantir.github.io/tslint/rules/typedef/) I see they have an option to ignore variable declarations for a function.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.18.0` |
| `@typescript-eslint/parser`        | `2.18.0` |
| `TypeScript`                       | `3.7.5` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `13.7.0` |
| `npm`                              | `6.13.6` |
